File: __iOS_BACKUP__/ShapeByte/features/workout/history/ui/WorkoutHistoryEntry.swift
//
//  WorkoutHistoryEntry.swift
//  ShapeByte
//
//  Created by Lang, Stefan [Shape Byte Tech] on 02.08.24.
//

import Foundation

struct WorkoutHistoryEntry: Equatable, Identifiable {
    var id: String {
        return entry.id
    }

    let entry: WorkoutScheduleEntry

    var name: String {
        return entry.name
    }

    var dateString: String {
        return DateStringFormatter.string(from: entry.date)
    }
}

File: features/workout/api/src/commonMain/kotlin/de/stefan/lang/shapebyte/features/workout/api/history/WorkoutHistoryEntry.kt
package de.stefan.lang.shapebyte.features.workout.api.history

import de.stefan.lang.foundation.core.contract.image.ImageResource
import de.stefan.lang.foundation.core.contract.stringformatter.DateTimeStringFormatter
import de.stefan.lang.shapebyte.features.workout.api.schedule.WorkoutScheduleEntry

data class WorkoutHistoryEntry(
    val entry: WorkoutScheduleEntry,
    private val dateStringFormatter: DateTimeStringFormatter,
) {
    val id: String by lazy { entry.id }
    val name: String by lazy { entry.name }

    val dateString: String by lazy {
        dateStringFormatter.formatDateToDDMMYYYY(entry.date)
    }

    // TODO: receive from WorkoutScheduleEntry / Workout
    val image: ImageResource by lazy { ImageResource(id = "Sprints.png") }
}
